tracer/src/Datadog.Trace/Configuration/TracerSettings.cs

```diff
@@ -1032,7 +1043,7 @@ public ImmutableTracerSettings Build()
         {
             var configurationDictionary = config
                    .WithKeys(key)
-                   .AsDictionary(allowOptionalMappings: true);
+                   .AsDictionary(allowOptionalMappings: true, () => new Dictionary<string, string>());
```

Comment:
This one is fine to default to `null` since there's check in the next line below.

Reply:
We could change that, but we don't currently support null as default value. We consider null as "no default value" and therefore don't record it for telemetry.

Reply:
Oh, ok.

Reply:
This means now though that `InitializeHeaderTags` never returns `null`... Should we change the subsequent check to check for an empty dictionary and return null, to avoid the allocation + (potential) change in behaviour?